libs/fluid-elements/virtual-scroll-container/src/lib/virtual-scroll-container.ts

```diff
@@ -0,0 +1,582 @@
+    this._cachedItems.get(itemId)!.height = item.getBoundingClientRect().height;
+  }
+
+  /**
+   * Observes intersections of the middle item of the scroll buffer
+   * As soon as it is half visible, another batch of items is rendered
+   */
+  private _observeVirtualScrollPadding(): void {
+    // Get elements to observe and add them to the observer
+    const observedItemStart = this._scrollContainer.querySelector(
+      `.fluid-virtual-scroll-item[data-index="${this._scrollState.bufferStartObservedItemIndex}"]`,
+    );
+
+    if (observedItemStart) {
+      this._virtualScrollBufferObserver.observe(observedItemStart);
+    }
+
+    const observedItemEnd = this._scrollContainer.querySelector(
+      `.fluid-virtual-scroll-item[data-index="${this._scrollState.bufferEndObservedItemIndex}"]`,
+    );
+
+    if (observedItemEnd) {
+      this._virtualScrollBufferObserver.observe(observedItemEnd);
+    }
+  }
+
+  private _handleHostIntersection(
+    observedItems: IntersectionObserverEntry[],
+  ): void {
+    observedItems.forEach((entry) => {
+      this._isVisible = entry.isIntersecting;
+      this.requestUpdate();
+    });
+  }
+
+  /**
+   * Handles intersection of items observed for updating the virtual scroll buffer
+   * @param observedItems List of observed items whose intersection ratio changed
+   */
+  private _handlePaddingIntersection(
+    observedItems: IntersectionObserverEntry[],
+  ): void {
+    observedItems.forEach((entry) => {
+      const item = entry.target as HTMLDivElement;
+      const itemIndex = parseInt(item.dataset.index!);
+
+      // Don't handle intersection triggered when adding an observed item
+      // except when the item has already passed the position where it is normally
+      // intersecting for the first time (end observed item at the bottom/right edge of the container,
+      // start observed item at the top/left edge)
+      // This might happen when the user is scrolling very fast and the observer
+      // calls the callback only when the item is already past the bottom/right edge of the
+      // container when scrolling downwards or the top/left edge when scrolling upwards
+      if (!this._observedItemsIndices.has(itemIndex)) {
+        if (
+          itemIndex === this._scrollState.bufferEndObservedItemIndex &&
+          item.getBoundingClientRect().bottom -
+            this._cachedItems.get(this.trackByFn(this._items[itemIndex]))!
+              .height *
+              VIRTUAL_SCROLL_THRESHOLD <=
+            entry.rootBounds!.bottom
+        ) {
+          // Observed item is at the end of the list, the item is already above the
+          // bottom edge of the container, hence the observer would not trigger again
+          // when scrolling further down
+          this._clearVirtualScrollBufferObserver();
+          this._scrollState.updateEndScrollBuffer();
+          this._renderVirtualScrollItems();
+          this._updatePlaceholders();
+        } else if (
```

Comment:
Instead of all the `else if` statements try to use early exists as it makes the code way more readable. Otherwise, you get a deep nesting and you don't need in which part of the if you are.

Reply:
I didn't do a lot of cleanup for this one, since this is most certainly the problem when it comes to the container stopping to render items and will be subject to change.